### ISSUE TITLE
Build fails when building against the 10.12 Mac OS X SDK with error error: use of undeclared identifier 'cupsGetPPD'	

### ISSUE THREAD

**Issue by sisufo** [electron/libchromiumcontent#241]
##### OS: 
macOS 10.12

##### Reproduce:
1. checkout source code of libchromiumcontent;
2. build it by instruction until the last step `script/build -t x64`

##### Error Message:
```
ninja: Entering directory `src/out/Release'
[1416/7457] CXX obj/printing/backend/printing.print_backend_cups.o
FAILED: ../../third_party/llvm-build/Release+Asserts/bin/clang++ -MMD -MF obj/printing/backend/printing.print_backend_cups.o.d -DU_COMBINED_IMPLEMENTATION -DUNISTR_FROM_CHAR_EXPLICIT= -DUNISTR_FROM_STRING_EXPLICIT= -DU_NO_DEFAULT_INCLUDE_UTF_HEADERS=0 -DV8_DEPRECATION_WARNINGS -D__ASSERT_MACROS_DEFINE_VERSIONS_WITHOUT_UNDERSCORE=0 -DDISABLE_NACL -DCHROMIUM_BUILD -DCR_CLANG_REVISION=274142-1 -DUSE_LIBJPEG_TURBO=1 -DENABLE_WEBRTC=1 -DENABLE_MEDIA_ROUTER=1 -DUSE_PROPRIETARY_CODECS -DENABLE_PEPPER_CDMS -DENABLE_NOTIFICATIONS -DUSE_EXTERNAL_POPUP_MENU -DFIELDTRIAL_TESTING_ENABLED -DENABLE_TASK_MANAGER=1 -DENABLE_EXTENSIONS=1 -DENABLE_PDF=1 -DENABLE_PLUGIN_INSTALLATION=1 -DENABLE_PLUGINS=1 -DENABLE_SESSION_SERVICE=1 -DENABLE_THEMES=1 -DENABLE_PRINTING=1 -DENABLE_BASIC_PRINTING=1 -DENABLE_PRINT_PREVIEW=1 -DENABLE_SPELLCHECK=1 -DUSE_BROWSER_SPELLCHECKER=1 -DENABLE_CAPTIVE_PORTAL_DETECTION=1 -DENABLE_SUPERVISED_USERS=1 -DENABLE_SERVICE_DISCOVERY=1 -DV8_USE_EXTERNAL_STARTUP_DATA -DFULL_SAFE_BROWSING -DSAFE_BROWSING_CSD -DSAFE_BROWSING_DB_LOCAL -DPRINTING_IMPLEMENTATION -DPRINT_BACKEND_AVAILABLE -DSK_SUPPORT_GPU=1 -DSK_IGNORE_DW_GRAY_FIX -DSK_IGNORE_LINEONLY_AA_CONVEX_PATH_OPTS -DU_USING_ICU_NAMESPACE=0 -DU_ENABLE_DYLOAD=0 -DU_NOEXCEPT= -DUSE_CUPS -DUSE_LIBPCI=1 -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -DNDEBUG -DNVALGRIND -DDYNAMIC_ANNOTATIONS_ENABLED=0 -D_FORTIFY_SOURCE=2 -Igen -I../.. -I../../skia/config -I../../third_party/skia/include/core -I../../third_party/skia/include/effects -I../../third_party/skia/include/pdf -I../../third_party/skia/include/gpu -I../../third_party/skia/include/lazy -I../../third_party/skia/include/pathops -I../../third_party/skia/include/pipe -I../../third_party/skia/include/ports -I../../third_party/skia/include/utils -I../../third_party/skia/include/utils/mac -I../../skia/ext -I../../third_party/icu/source/i18n -I../../third_party/icu/source/common -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.sdk -O2 -gdwarf-2 -fvisibility=hidden -Werror -mmacosx-version-min=10.8 -arch x86_64 -Wno-deprecated-declarations -Wall -Wextra -Wno-unused-parameter -Wno-missing-field-initializers -Wno-selector-type-mismatch -Wpartial-availability -Wheader-hygiene -Wno-char-subscripts -Wno-unneeded-internal-declaration -Wno-covered-switch-default -Wstring-conversion -Wno-c++11-narrowing -Wno-deprecated-register -Wno-inconsistent-missing-override -Wno-shift-negative-value -Wno-undefined-var-template -Wno-nonportable-include-path -std=c++11 -stdlib=libc++ -fno-rtti -fno-exceptions -fvisibility-inlines-hidden -fno-threadsafe-statics -fcolor-diagnostics -Xclang -load -Xclang /Users/sisufo/Workspace/electron/source-code/libchromiumcontent/src/third_party/llvm-build/Release+Asserts/lib/libFindBadConstructs.dylib -Xclang -add-plugin -Xclang find-bad-constructs -Xclang -plugin-arg-find-bad-constructs -Xclang check-templates -Xclang -plugin-arg-find-bad-constructs -Xclang follow-macro-expansion -Xclang -plugin-arg-find-bad-constructs -Xclang check-implicit-copy-ctors -fno-strict-aliasing  -c ../../printing/backend/print_backend_cups.cc -o obj/printing/backend/printing.print_backend_cups.o
../../printing/backend/print_backend_cups.cc:243:21: error: use of undeclared identifier 'cupsGetPPD'
    ppd_file_path = cupsGetPPD(name);
                    ^
../../printing/backend/print_backend_cups.cc:256:21: error: use of undeclared identifier 'cupsGetPPD2'
    ppd_file_path = cupsGetPPD2(http.http(), name);
                    ^
2 errors generated.
[1416/7457] CXX obj/services/shell/public/cpp/lib/shell_public.capabilities.o
ninja: build stopped: subcommand failed.
Traceback (most recent call last):
  File "script/build", line 52, in <module>
    sys.exit(main())
  File "script/build", line 38, in main
    subprocess.check_call([NINJA, '-C', config_dir, target])
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 540, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['/Users/sisufo/Workspace/electron/source-code/libchromiumcontent/vendor/depot_tools/ninja', '-C', 'src/out/Release', 'chromiumcontent_all']' returned non-zero exit status 1
```

##### Clue：
[same issue in chromium tracker](https://bugs.chromium.org/p/chromium/issues/detail?id=648017&can=1&q=cupsGetPPD&colspec=ID%20Pri%20M%20Stars%20ReleaseBlock%20Component%20Status%20Owner%20Summary%20OS%20Modified)

##### Question:
Since the problem is caused by lower version of chromium, when higher chromium can be supported, thanks.

**Comment by bbondy** [comment]
This is a problem with Xcode 8.0.

As a workaround you can install the old version side by side and make it use that.

First check your version:

> xcodebuild -version

It'll probably show 8.0. 

You can download Xcode 7.3.1 here:
https://developer.apple.com/download/more/

After it is downloaded install it side by side to `/Applications/Xcode_7.3.1.app/`

Then run:

> sudo xcode-select -s /Applications/Xcode_7.3.1.app/

**Comment by poiru** [comment]
Xcode 8 build failures have been fixed in:

* https://chromium.googlesource.com/chromium/src.git/+/29c1262dbd4f749dd514d05a571cdd809ba8cfe0
* https://chromium.googlesource.com/chromium/src.git/+/913febd51b3506b59e9dd06ffd2869837f4bfc14

I suggest applying those patches locally for now.

**Comment by kevinsawicki** [comment]
Both patches mentioned by @poiru have now landed on master via the Chrome 54 upgrade in #254 

This issue should be resolved now.
